src/store/actions/api.actions.js

```diff
@@ -21,81 +23,115 @@ const history = createHistory({
   forceRefresh: true,
 });
 
-export const getAPIRequest = () => ({
-  type: FETCH_API_REQUEST,
+export const deleteEndpointRequest = () => ({
+  type: DELETE_ENDPOINT_START,
 });
 
-export const getAPISuccess = api => ({
-  type: FETCH_API_SUCCESS,
+export const deleteEndpointSuccess = () => ({
+  type: DELETE_ENDPOINT_SUCCESS,
+});
+
+export const getEndpointRequest = () => ({
+  type: FETCH_ENDPOINT_START,
+});
+
+export const getEndpointSuccess = api => ({
+  type: FETCH_ENDPOINT_SUCCESS,
   payload: api,
 });
 
-export const getApiSchemaRequest = () => ({
-  type: FETCH_API_SCHEMA_REQUEST,
+export const getEndpointSchemaRequest = () => ({
+  type: FETCH_ENDPOINT_SCHEMA_START,
 });
 
-export const getApiSchemaSuccess = api => ({
-  type: FETCH_API_SCHEMA_SUCCESS,
+export const getEndpointSchemaSuccess = api => ({
+  type: FETCH_ENDPOINT_SCHEMA_SUCCESS,
   payload: api,
 });
 
-export const saveAPIRequest = () => ({
-  type: SAVE_API_REQUEST,
+export const saveEndpointRequest = () => ({
+  type: SAVE_ENDPOINT_START,
 });
 
-export const saveAPISuccess = api => ({
-  type: SAVE_API_SUCCESS,
+export const saveEndpointSuccess = api => ({
+  type: SAVE_ENDPOINT_SUCCESS,
   payload: api,
 });
 
-export const resetAPI = () => ({
-  type: RESET_API,
+export const resetEndpoint = () => ({
+  type: RESET_ENDPOINT,
 });
 
 export const willClone = data => ({
   type: WILL_CLONE,
   payload: data,
 });
 
-export const fetchAPI = (pathname) => dispatch => {
-  dispatch(getAPIRequest());
-  
+export const deleteEndpoint = (apiName, callback) => async dispatch => {
+  console.log('CALLBACK:: ', callback);
+  dispatch(deleteEndpointRequest());
+
+  try {
+    const response = await client.delete(`apis/${apiName}`);
+
+    dispatch(deleteEndpointSuccess());
+    dispatch(openResponseModal({
```

Comment:
so does it make sense to dispatch additional action? Maybe it's simpler to react on `deleteEndpointSuccess` in reducer